src/ncp/ncp_base_mtd.cpp

```diff
@@ -2975,6 +2977,22 @@ template <> otError NcpBase::HandlePropertySet<SPINEL_PROP_CNTR_ALL_IP_COUNTERS>
     return OT_ERROR_NONE;
 }
 
+template <> otError NcpBase::HandlePropertySet<SPINEL_PROP_STREAM_CLI>(void)
+{
+    otError     error  = OT_ERROR_NONE;
+    const char *string = nullptr;
+    size_t      length = 0;
+
+    SuccessOrExit(error = mDecoder.ReadUtf8(string));
+    length = strlen(string);
+    VerifyOrExit(length < OPENTHREAD_CONFIG_CLI_MAX_LINE_LENGTH, error = OT_ERROR_INVALID_COMMAND);
```

Comment:
The maximum input line length is already checked by the CLI module itself. It uses `StringLength()`, which is safer than `strlen()` (in the sense that it checks up to a max length and works correctly in case the string is not properly null-terminated). I suggest removing this extra check here in `NcpBase`:

https://github.com/openthread/openthread/blob/a8fef54bf65c11be7d3ae276d4f86f291b304ba0/src/cli/cli.cpp#L291-L309